File: src/main/java/com/nicholasnassar/sfsapi/data/assignments/FullAssignment.java
package com.nicholasnassar.sfsapi.data.assignments;

import com.nicholasnassar.sfsapi.data.links.Resource;
import io.vertx.core.json.JsonObject;

import java.util.List;

public class FullAssignment {
    private final String activity, clazz, instructor, assigned, due, possiblePoints, category, notes;

    private final List<Resource> resources;

    public FullAssignment(String activity, String clazz, String instructor, String assigned, String due,
                          String possiblePoints, String category, String notes, List<Resource> resources) {
        this.activity = activity;

        this.clazz = clazz;

        this.instructor = instructor;

        this.assigned = assigned;

        this.due = due;

        this.possiblePoints = possiblePoints;

        this.category = category;

        this.notes = notes;

        this.resources = resources;
    }

    public JsonObject asJson() {
        return new JsonObject().put("activity", activity).put("class", clazz).put("instructor", instructor)
                .put("assigned", assigned).put("due", due).put("possible_points", possiblePoints)
                .put("category", category).put("notes", notes).put("resources", Resource.asJsonArray(resources));
    }
}

File: src/main/java/com/nicholasnassar/sfsapi/data/assignments/FullAssignment.kt
package com.nicholasnassar.sfsapi.data.assignments

import com.nicholasnassar.sfsapi.data.links.Resource
import io.vertx.core.json.JsonObject

class FullAssignment(val activity: String, val clazz: String?, val instructor: String?, val assigned: String?,
                     val due: String?, val possiblePoints: String?, val category: String?, val notes: String?,
                     val resources: List<Resource>) {
    fun asJson(): JsonObject {
        return JsonObject().put("activity", activity).put("class", clazz).put("instructor", instructor)
                .put("assigned", assigned).put("due", due).put("possible_points", possiblePoints)
                .put("category", category).put("notes", notes).put("resources", Resource.asJsonArray(resources))
    }
}
